app/src/main/java/org/mozilla/focus/components/RecursiveFileObserver.java

```diff
@@ -0,0 +1,77 @@
+package org.mozilla.focus.components;
+
+import android.os.FileObserver;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
+public class RecursiveFileObserver extends FileObserver {
```

Comment:
I should have recalled that FileObservers are not recursive before saying that my proposal is the easy way... sorry.
This does not work on my LG and I believe doing this recursively on every folder is not a good idea. I'm thinking of listening to MediaStore.Images.Media.EXTERNAL_CONTENT_URI with a ContentObserver. What do you think, @idiotmax ?

Reply:
We're actually implementing screenshot observing with the same concept on Fennec, amazing! https://dxr.mozilla.org/mozilla-central/source/mobile/android/base/java/org/mozilla/gecko/ScreenshotObserver.java Will dig deeper tomorrow.

Reply:
I've checked on my LG with the old single folder implementation, it does not work. It's not sending the CREATE event, but only OPEN, ACCESS, CLOSE_NOWRITE. My opinion is still toward changing the architecture since I've heard some unreliability on FileObserver after Marshmallow.